cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Interfaces.kt

```diff
@@ -149,3 +149,11 @@ interface HasOverloadedOperation : HasOperatorCode {
      */
     val operatorBase: Expression
 }
+
+interface HasAccess {
+    /**
+     * Is this reference used for writing data instead of just reading it? Determines dataflow
+     * direction
+     */
+    var access: AccessValues
```

Comment:
On second thought. Wouldn't it make sense to just implements this on all `Expression` nodes? The memory overhead is probably negligible and we would remove a lot of ugly `as? HasAccess`  casts. Basically this should apply to all expressions.

Reply:
I decided against this because many `Expression`s cannot receive any data (i.e., `AccessValues.WRITE` never makes sense for them and for some, also `READ` might be confusing), so the access value is not useful there. Are the casts are a major concern? Any opinions on this @konradweiss ?

Reply:
But all of them (most?) are part of the DFG and it is interesting to have a harmonized way to access this information even for a lot of them it will statically stay with READ.

Reply:
Probably altough a little bit awkward :(

Reply:
Well, I kept it with Expressions for now and removed the interface.

Reply:
Sounds like a reasonable compromise